src/osd/ReplicatedPG.cc

```diff
@@ -13201,8 +13203,14 @@ boost::statechart::result ReplicatedPG::TrimmingObjects::react(const SnapTrim&)
     }
 
     dout(10) << "TrimmingObjects react trimming " << pos << dendl;
-    OpContextUPtr ctx = pg->trim_object(pos);
-    if (!ctx) {
+    OpContextUPtr ctx;
+    r = pg->trim_object(pos, &ctx);
+    if (r == -ENOENT) {
+      dout(10) << "TrimmingObjects cannot find snap, dropping from snaptrimq: " << pos << dendl;
+      post_event(SnapTrim());
```

Comment:
you might want to check `in_flight.empty()` before transiting to `WaitingOnReplicas`, if it's already empty, should move to `NotTrimming` instead.

Reply:
Alright, I updated the change – is that what you meant? Compiles for me and doesn't explode right away ... /o\